File: my_school_frontend_clean/lib/models/exam_grade.dart
class ExamGrade {
  final String studentId;
  final String studentName;
  final double grade;
  final String appreciation;
  final String? photoUrl;
  final DateTime submittedAt;

  ExamGrade({
    required this.studentId,
    required this.studentName,
    required this.grade,
    required this.appreciation,
    this.photoUrl,
    required this.submittedAt,
  });

  Map<String, dynamic> toJson() => {
    'studentId': studentId,
    'studentName': studentName,
    'grade': grade,
    'appreciation': appreciation,
    'photoUrl': photoUrl,
    'submittedAt': submittedAt.toIso8601String(),
  };

  factory ExamGrade.fromJson(Map<String, dynamic> json) => ExamGrade(
    studentId: json['studentId'],
    studentName: json['studentName'],
    grade: json['grade'].toDouble(),
    appreciation: json['appreciation'] ?? '',
    photoUrl: json['photoUrl'],
    submittedAt: DateTime.parse(json['submittedAt']),
  );
}

File: my-school-backend/src/models/ExamGrade.js
const mongoose = require('mongoose');

const examGradeSchema = new mongoose.Schema({
  examId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'AgendaEvent',
    required: true
  },
  studentId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Student',
    required: true
  },
  studentName: {
    type: String,
    required: true
  },
  subject: {
    type: String,
    required: true
  },
  grade: {
    type: Number,
    required: true,
    min: 0,
    max: 20
  },
  appreciation: {
    type: String,
    default: ''
  },
  photoUrl: {
    type: String,
    default: null
  },
  teacherId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Teacher',
    required: true
  },
  teacherName: {
    type: String,
    required: true
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

examGradeSchema.index({ examId: 1, studentId: 1 }, { unique: true });
examGradeSchema.index({ studentId: 1 });

module.exports = mongoose.model('ExamGrade', examGradeSchema);